### ISSUE TITLE
.remove_background signal_range should have different behaviour on ints and floats

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1267]
With `s.remove_background()` one can specify the signal range to base the background on, using the argument `signal_range=tuple`.
I think the tuple should have the same behaviour as s.isig[], that is, that if the range is given in floats it should refer to the calibrated signal axis, and if given in ints it should refer to the index.

It currently refers to the ~~index~~axis.

**Comment by francisco-dlp** [comment]
+1. This is a common problem: the way to specify ranges is not consisted across different methods (not only remove_background) in hyperspy.

**Comment by thomasaarholt** [comment]
I'm really confused now. signal_range seems to happily accept either ints and floats. I suspect it actually only takes floats - that is, it only takes values already calibrated to the signal axis.

**But it does not throw an error** if the entered values are outside the range, which had me super confused to start with.

**Comment by francisco-dlp** [comment]
Yes, it takes the range in the axis units. And yes, it should throw an error when the range is fully out of the axis interval.
